app/src/main/java/de/westnordost/streetcomplete/quests/seating/AddSeating.kt

```diff
@@ -0,0 +1,61 @@
+package de.westnordost.streetcomplete.quests.seating
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement
+
+class AddSeating : OsmFilterQuestType<Seating>() {
+    override val elementFilter = """
+        nodes, ways with amenity ~ restaurant|cafe|fast_food|ice_cream|food_court|pub|bar
+        and !outdoor_seating
+    """
+    override val changesetComment = "Add seating info"
+    override val defaultDisabledMessage = R.string.default_disabled_msg_summer_outdoor_seating
+    override val wikiLink = "Key:outdoor_seating"
+    override val icon = R.drawable.ic_seating_quest
+    override val isReplaceShopEnabled = true
+
+    override val questTypeAchievements = listOf(QuestTypeAchievement.CITIZEN)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_seating_name_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("""
+            nodes, ways with
+            amenity = bar
+            or amenity = pub
+            or amenity = food_court
+            or amenity = ice_cream
+            or amenity = restaurant
+            or amenity = cafe
+            or amenity = fast_food
+        """)
+
+    override fun createForm() = AddSeatingForm()
+
+    override fun applyAnswerTo(answer: Seating, tags: Tags, timestampEdited: Long) {
+        when (answer) {
+            Seating.NO -> {
+                tags["takeaway"] = "only"
+                tags["outdoor_seating"] = "no"
+                tags["indoor_seating"] = "no"
+            }
+            Seating.ONLY_OUTDOOR -> {
+                tags["outdoor_seating"] = "yes"
+                tags["indoor_seating"] = "no"
+            }
+            Seating.ONLY_INDOOR -> {
+                tags["outdoor_seating"] = "no"
+                tags["indoor_seating"] = "yes"
+            }
+            Seating.INDOOR_AND_OUTDOOR -> {
+                tags["outdoor_seating"] = "yes"
+                tags["indoor_seating"] = "yes"
+            }
+        }
```

Comment:
Using the `enum class` extension suggestion from my previous comment, this can be simplified to:

```suggestion
        if (answer == Seating.NO) {
            tags["takeaway"] = "only"
        }
        
        tags["outdoor_seating"] = answer.hasOutdoorSeating.toYesNo()
        tags["indoor_seating"] = answerhasIndoorSeating.toYesNo()
```

(You might need to import the `toYesNo` extension function though.)